AMDirT/core/__init__.py

```diff
@@ -190,6 +216,59 @@ def prepare_eager_table(
     return selected_libraries
 
 
+@st.cache()
+def prepare_mag_table(
+    samples: pd.DataFrame,
+    libraries: pd.DataFrame,
+    table_name: str,
+    supported_archives: Iterable[str],
+) -> pd.DataFrame:
+    """Prepare nf-core/mag tsv input table
+
+    Args:
+        sample (pd.dataFrame): selected samples table
+        library (pd.dataFrame): library table
+        table_name (str): Name of the table
+        supported_archives (list): list of supported archives
+    """
+    stacked_samples = (
+        samples.query("archive in @supported_archives")
+        .loc[:, "archive_accession"]
+        .str.split(",", expand=True)
+        .stack()
+        .reset_index(level=0)
+        .set_index("level_0")
+        .rename(columns={0: "archive_accession"})
+        .join(samples.drop("archive_accession", axis=1))
+    )
+
+    if table_name in [
+        "ancientmetagenome-environmental",
+    ]:
+        sel_col = ["archive_accession"]
+    else:
+        sel_col = ["archive_accession", "sample_host"]
```

Comment:
Is there a reason why sample_host is relevent here?

Reply:
@jfy133 I agree, It's useless. I removed it.